### PR TITLE
RDSS-807

### PR DESCRIPTION
**Pull request by techjacker**
Taxonomy Definitions Update Script

### CODE REVIEW

File: datamodel/APC.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "APC",
+  "description": "Article Processing Charge status",
+  "type": "object",
+  "properties": {
+    "id": {
+      "description": "The unique identifier for the vocabulary",
+      "default": 1,
```
Comment (danielgrant) [inline datamodel/APC.json]: Would suggest removing these `default` properties from all the JSON Schema files, they serve no purpose and their meaning can confuse.

File: datamodel/APC.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      "type": "integer"
+    },
+    "vocabulary": {
+      "description": "The APC",
+      "type": "string",
+      "enum": ["paid", "partiallyWaived", "fullyWaived", "notCharged", "notRequired", "unknown"]
+    }
+  },
+  "required": ["id", "vocabulary"]
+}
```
Comment (danielgrant) [inline datamodel/APC.json]: Suggest the JSON Schema files are tightened up through use of `"additionalProperties": false`, preventing additional properties from leaking into requests without being specifically added to the JSON Schema files.

**Review by danielgrant — CHANGES_REQUESTED** [review]
Only a couple of comments around the JSON Schema files from my end.

File: datamodel/APC.json (hunk 1 above)
Comment (techjacker) [reply datamodel/APC.json]: We're using the default id as the database IDs when updating the service.

Comment (danielgrant) [reply datamodel/APC.json]: I can see now how the `value` fields are being used. I would suggest that we reconsider this approach - embedding values like this in JSON Schema which are then copied into the JSON payload that's sent to the API is not what JSON Schema is designed for.

In addition to this, the files under `examples/...` all include the `value` attribute. If we choose to keep the `value` attribute in the JSON Schema documents as it is now, I would expect these example JSON files to not include the `value` attribute.

**Review by danielgrant — APPROVED** [review]

**Comment by danielgrant** [comment]
Hi @techjacker, these changes look good. I'm happy for the pull request to be merged.

**Pull request merged**